source-code/sdk-js/src/runtime/runtime.test.ts

```diff
@@ -0,0 +1,129 @@
+import { describe, expect, test } from "vitest"
+import { initBaseRuntime, initRuntime, RuntimeState } from "./runtime.js"
+
+describe("initRuntime", () => {
+	test("it should provide all functions", () => {
+		const runtime = initRuntime()
+
+		expect(runtime.loadResource).toBeDefined()
+		expect(runtime.switchLanguage).toBeDefined()
+		expect(runtime.getLanguage).toBeDefined()
+		expect(runtime.getLookupFunction).toBeDefined()
+	})
+})
+
+describe("initBaseRuntime", () => {
+	test("it should provide all functions", () => {
+		const runtime = initBaseRuntime()
+
+		expect(runtime.loadResource).toBeDefined()
+		expect(runtime.switchLanguage).toBeDefined()
+		expect(runtime.getLanguage).toBeDefined()
+		expect(runtime.getLookupFunction).toBeDefined()
+	})
+
+	describe("loadResource", () => {
+		// TODO: implement `loadResource`
+		test.todo("it should load a resource")
+	})
+
+	describe("switchLanguage", () => {
+		test("it should switch the language", () => {
+			const state = {
+				language: "en",
+				resources: new Map(),
+			} satisfies RuntimeState
+
+			const runtime = initBaseRuntime(state)
+
+			expect(state.language).toBe("en")
+
+			runtime.switchLanguage("fr")
+
+			expect(state.language).toBe("fr")
+		})
+	})
+
+	describe("getLanguage", () => {
+		test("it should return undefined if language was never set", () => {
+			const runtime = initBaseRuntime()
+
+			expect(runtime.getLanguage()).toBeUndefined()
+		})
+
+		test("it should return the current language", () => {
+			const state = {
+				language: "en",
+				resources: new Map(),
+			} satisfies RuntimeState
+
+			const runtime = initBaseRuntime(state)
+
+			state.language = "de"
+
+			expect(runtime.getLanguage()).toBe("de")
+		})
+	})
+
+	describe("getLookupFunction", () => {
+		test("it should not throw if language was never set", () => {
+			const runtime = initBaseRuntime()
+
+			const i = runtime.getLookupFunction()
+
+			expect(i("test")).toBe("")
```

Comment:
I am not sure if I understand why this shouldn't throw (or at least how this should be an empty string)

Reply:
You probably won't crash your whole application if you reference a message that is not present. Rendering an empty string is the safest way to not affect anything else on screen.